app/guides/utils.ts

```diff
@@ -0,0 +1,96 @@
+"use server";
+import { ReturnDocument } from "../models/return";
+import {
+  FeedbackDocument,
+  ReviewedFeedbackDocument,
+  Vote,
+} from "../models/review";
+import {
+  ReturnStatus,
+  FeedbackStatus,
+  ReviewsReceivedStatus,
+  ReviewsGivenStatus,
+} from "./types";
+
+export const calculateReturnStatus = (
+  returnedGuides: ReturnDocument[],
+  feedbackReceived: FeedbackDocument[]
+): ReturnStatus => {
+  if (returnedGuides.length === 0) {
+    return ReturnStatus.NOT_RETURNED;
+  }
+
+  const latestSubmission = returnedGuides.reduce(
+    (latestSubmission, submission) => {
+      if (submission.createdAt > latestSubmission.createdAt) {
+        return submission;
+      }
+      return latestSubmission;
+    }
+  );
+  const feedbackOnLatestSubmission = feedbackReceived.filter((feedback) => {
+    return feedback.return === latestSubmission._id;
+  });
+
+  if (
+    feedbackOnLatestSubmission.filter(
+      (feedback) => feedback.vote === Vote.NO_PASS
+    ).length > 0
+  ) {
+    return ReturnStatus.FAILED;
+  }
+  if (feedbackOnLatestSubmission.length < 2) {
+    return ReturnStatus.AWAITING_FEEDBACK;
+  }
+
+  return ReturnStatus.PASSED;
+};
```

Comment:
There is missing the "recommend to all of fame" (the purple star in the figma document).

Reply:
Does it automatically go to the hall of fame if 1 person votes it that way?